### PR TITLE
upgrades: fix the write of 'version' to system.tenant_settings

### PR DESCRIPTION
**Pull request by knz**
Prior to this patch, a bogus value was written to the tenant_id column for the 'version' override. This patch fixes it.

We also suspect this will help with the multitenant-upgrade roachtest but this will be checked in a later step.

Informs #105858.
Release note: None
Epic: CRDB-26691

### CODE REVIEW

**Review by ajstorm — APPROVED** [review]
I'm not sure if this PR has been superseded, but if it's still valid, it LGTM (modulo the testing comment).

**Comment by knz** [comment]
>  if it's still valid, it LGTM

This PR has a fundamental flaw: it doesn't do anything to "fix" clusters that have run the migration already.
Especially for the 23.1 branch we cannot backport a new version gate + migration. It's unclear how to make 23.1 run the code that would re-create the faulty row in tenant_settings.

**Comment by ajstorm** [comment]
>It's unclear how to make 23.1 run the code that would re-create the faulty row in tenant_settings.

Wouldn't this only impact UA and/or Serverless clusters created on 23.1?  If so, we know that there shouldn't be any UA clusters created (yet), and we should be able to track down all of the Serverless ones too.  If we backport this fix to 23.1 and ensure that all C2C customers wait for that patch release for their testing, the UA case should be good.  For the Serverless case, can we not patch all of those manually and not worry about a code fix?

Actually, thinking about this a bit more, we might be OK regardless.  The override was meant to prevent tenants from upgrading if the upgrade will push their version beyond that of the storage cluster.  If the override is unset (as is the case for newly created 23.1 clusters, due to the bug, and all pre-23.1 clusters) the tenant interprets that cluster as 22.2.  This will prevent a tenant created on 23.1 from upgrading as it thinks that it's already a release ahead of the storage cluster, which is the correct behaviour (though the error message on upgrade is bad).  What's important to note however, is that when a 23.1 storage cluster is upgraded, the override _will_ be set correctly.  As a result, upgrading the storage cluster to 23.2 should automatically fix this problem for all existing clusters.  

We'd want to test this to confirm, but I suspect we'll be OK here without a code fix for the bad setting in 23.1.

**Comment by knz** [comment]
Let's go ahead and merge+backport this one.
It does not preclude us from moving forward with #106037 separately/afterwards; at least the proximate issue would be fixed.

bors r=ajstorm

**Pull request merged**
